src/vm/memory/relocatable.zig

```diff
@@ -537,6 +745,26 @@ test "MaybeRelocatable: tryIntoU64 should return an error if MaybeRelocatable Fe
     try expectError(error.ValueTooLarge, maybeRelocatable.tryIntoU64());
 }
 
+test "MaybeRelocatable: any comparision should return false if other MaybeRelocatable is of different variant 1" {
+    var maybeRelocatable1 = MaybeRelocatable{ .relocatable = Relocatable.new(0, 10) };
+    var maybeRelocatable2 = MaybeRelocatable{ .felt = Felt252.fromInteger(10) };
+
+    try expect(!maybeRelocatable1.lt(maybeRelocatable2));
+    try expect(!maybeRelocatable1.le(maybeRelocatable2));
+    try expect(!maybeRelocatable1.gt(maybeRelocatable2));
+    try expect(!maybeRelocatable1.ge(maybeRelocatable2));
+}
+
+test "MaybeRelocatable: any comparision should return false if other MaybeRelocatable is of different variant 2" {
+    var maybeRelocatable1 = MaybeRelocatable{ .felt = Felt252.fromInteger(10) };
+    var maybeRelocatable2 = MaybeRelocatable{ .relocatable = Relocatable.new(0, 10) };
+
+    try expect(!maybeRelocatable1.lt(maybeRelocatable2));
+    try expect(!maybeRelocatable1.le(maybeRelocatable2));
+    try expect(!maybeRelocatable1.gt(maybeRelocatable2));
+    try expect(!maybeRelocatable1.ge(maybeRelocatable2));
+}
+
```

Comment:
Can we also test one true assertion for each `MaybeRelocatable` comparison case? Like:
- One `lt` between two `Relocatable`
- One `le` between two `Relocatable`
- One `gt` between two `Relocatable`
- One `ge` between two `Relocatable`
- One `lt` between two `Felt252`
- One `le` between two `Felt252`
- One `gt` between two `Felt252`
- One `ge` between two `Felt252`

Reply:
i thought about it but since when both variants are same, result is same as operation on that type directly so its already covered by other tests.

Reply:
Yes I know it's just to confirm that for example there is not a hidden typo in one of the variants which would call `ge` instead of `gt` or `lt`. But in absolute terms it's not a necessity. If it takes you a few minutes do it otherwise we'll do it later when we don't know what to do :)

Reply:
> Yes I know it's just to confirm that for example there is not a hidden typo
right that make sense, i will add them